prow/github/types.go

```diff
@@ -300,21 +300,85 @@ type PullRequestChange struct {
 // Repo contains general repository information.
 // See also https://developer.github.com/v3/repos/#get
 type Repo struct {
-	Owner         User   `json:"owner"`
-	Name          string `json:"name"`
-	FullName      string `json:"full_name"`
-	HTMLURL       string `json:"html_url"`
-	Fork          bool   `json:"fork"`
-	DefaultBranch string `json:"default_branch"`
-	Archived      bool   `json:"archived"`
-
+	Owner            User   `json:"owner"`
+	Name             string `json:"name"`
+	FullName         string `json:"full_name"`
+	HTMLURL          string `json:"html_url"`
+	Fork             bool   `json:"fork"`
+	DefaultBranch    string `json:"default_branch"`
+	Archived         bool   `json:"archived"`
+	Private          bool   `json:"private"`
+	Description      string `json:"description"`
+	Homepage         string `json:"homepage"`
+	HasIssues        bool   `json:"has_issues"`
+	HasProjects      bool   `json:"has_projects"`
+	HasWiki          bool   `json:"has_wiki"`
+	AllowSquashMerge bool   `json:"allow_squash_merge"`
+	AllowMergeCommit bool   `json:"allow_merge_commit"`
+	AllowRebaseMerge bool   `json:"allow_rebase_merge"`
 	// Permissions reflect the permission level for the requester, so
 	// on a repository GET call this will be for the user whose token
 	// is being used, if listing a team's repos this will be for the
 	// team's privilege level in the repo
 	Permissions RepoPermissions `json:"permissions"`
 }
 
+// RepoRequest contains metadata used in requests to create or update a Repo.
+// Compared to `Repo`, its members are pointers to allow the "not set/use default
+// semantics.
+// See also:
+// - https://developer.github.com/v3/repos/#create
+// - https://developer.github.com/v3/repos/#edit
+type RepoRequest struct {
+	Name             *string `json:"name,omitempty"`
```

Comment:
Why are all of these pointers? It doesn't follow the existing pattern. And I do not think it is important to be able to create a repo named `""`

Reply:
They are pointers to allow the user/caller to not set that given parameter (as opposed to set it to the empty value), because in this API most parameters are optional and their omission from the call usually means "use default". If they weren't pointers, they would either:
1. be `omitempty` and so it would not be possible to pass that parameter with e.g. and empty string or `false`
2. would not be `omitempty`, and so it would not be possible to not set such parameter

I guess for the `Name` specifically it could be a non-pointer, `omitempty` string, but for others `""` can be a legit value.

Reply:
For now I kept even `Name` as a pointer, I felt it would be confusing if it were the only non-pointer member. Let me know if you want me to change it.